rsts/howto/performance/index.rst

```diff
@@ -4,6 +4,90 @@
+
+.. list-table:: Important Properties
+   :widths: 25 25 25 50
+   :header-rows: 1
+
+   * - Property
+     - Section
+     - Rule of thumb
+     - Description
+   * - workers
+     - propeller
+     - larger the number, more the performance (not too high)
+     - Number of `logical threads` workers, that can work concurrently. Also implies number of workflows that can be executed in parallel. Since Flytepropeller uses go-routines, it can run way  more than number of physical cores.
+   * - workflow-reeval-duration
+     - propeller
+     - lower the number - lower latency, lower throughput
+     - frequency at which, given no external signal, a workflow should be re-evaluated by Flyte propellers reval loop
+   * - downstream-eval-duration
+     - propeller
+     - lower the number - lower latency and throughput
+     - This indicates how often are external events like pods completion etc recorded.
+   * - max-streak-length
+     - propeller
+     - higher the number lower the latency for end to end workflow, epsecially for cached workflows
+     - number of consecutive rounds to try with one workflow - prioritize a hot workflow over others.
+   * - kube-client-config
+     - propeller
+     - this is a very important config
+     - this configures the kubernetes client used by flytepropeller
+
+In the above table the 2 most important configs are ``workers`` and ``kube-client-config``.
+
+Kubeclient config is usually configured poorly and leads to bad performance. This is especially noticeable, if your workload involves spawning a lot of pods or other CRDs. For case in which your workload is a good mix of K8s local resources and external resources, the default for this configuration should suffice.
+Flytepropeller configures a default version, which is better than the actual default. This configuration is critical, as this improves the number of requests that Flyte can send to KubeAPI, which is critical. An example kube-client-config is as follows
+
+.. code-block:: yaml
+
+    propeller:
+        kube-client-config:
+            qps: 100 # Refers to max rate of requests to KubeAPI server
+            burst: 50 # refers to max burst rate to Kube API server
+            timeout: 30s # Refers to timeout when talking with kubeapi server
+
+
+.. note:: As you increase the number of workers in FlytePropeller it is important to increase the number of cpu's given to flytepropeller pod.
+
+
+Another area of slowdown could be the size of the input-output cache that Flytepropeller maintains in-memory. This can be configured, while configuring
+the storage for FlytePropeller.
+
+Rule of thumb, for Flytepropeller that has 2GB or memory, should allocate about 512MB-756MB to the input/output cache.
+
+
+Scaling out FlyteAdmin
```

Comment:
When should they scale vertically vs horizontally? maybe link to horizontal autoscaler? is there a good metric to use to scale up horizontally? Maybe request rate?